query-engine/driver-adapters/executor/src/bench.ts

```diff
@@ -0,0 +1,104 @@
+import * as qe from "./qe";
+
+// pg dependencies
+import pgDriver from "pg";
+import * as prismaPg from "@prisma/adapter-pg";
+
+import { DriverAdapter } from "@prisma/driver-adapter-utils";
+import { webcrypto } from "node:crypto";
+
+import { recordReplay } from "./recordReplay";
+import prismaQueries from "../bench/queries.json";
+
+(global as any).crypto = webcrypto;
+
+async function main(): Promise<void> {
+  // read the prisma schema from stdin
+  const prismaSchema = await new Promise<string>((resolve, reject) => {
+    let data = "";
+    process.stdin.on("data", (chunk) => {
+      data += chunk;
+    });
+    process.stdin.on("end", () => {
+      resolve(data);
+    });
+    process.stdin.on("error", reject);
+  });
+
+  const url = process.env.DATABASE_URL;
+  if (url == null) {
+    throw new Error("DATABASE_URL is not defined");
+  }
+  const pg = await pgAdapter(url);
+  const { recorder, replayer } = recordReplay(pg);
+
+  await recordQueries(recorder, prismaSchema, prismaQueries);
+  await benchMarkQueries(replayer, prismaSchema, prismaQueries);
```

Comment:
And then we benchmark them using a replayer, which is a decorated driver adapter that reads the results from an in memory store.